Plotly.Blazor.Examples/EventHelper.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+
+using Microsoft.AspNetCore.Components;
+
+namespace Plotly.Blazor.Examples
+{
+
+    public class XYClickEventArgs : EventArgs
+    {
+        public string X { get; set; }
+        public int Y { get; set; }
+    }
+
+    public class PieClickEventArgs : EventArgs
+    {
+        public string Label { get; set; }
+        public int Value { get; set; }
+    }
+    
+    public class PieHoverEventArgs : EventArgs
+    {
+        public string Label { get; set; }
```

Comment:
That's a interesting idea for getting the Pie trace. What type do the values in the Pie traces `IList<object> Values` come through as? I think when I tried defining `X` as object it comes through as a `JsonElement` which would need to then be converted to a string or number.

Personally I've also just exposed `curveNumber` and `pointNumber`, and can then reference the in memory `ITrace` from my components that is passed to the blazor chart.

> I think that by creating all of these PieHoverEventArgs etc types we are ditching quite a lot of data.

The design here is very open, `Plotly.Blazor` forwards all the event data, and the consuming project (e.g. Examples) decides what event handlers to define in JavaScript and C#. The example here intentionally only passed through data it was interested in to reduce the amount of data being transferred/deserialized and improve readability, but it could pass through a lot more as needed.
This makes it very flexible for different charts returning different data shapes / types (e.g. is `X` a string or number) but pushes the complexity of defining them the consuming developers. 
Being able to define some standard click handlers for the most common cases in `Plotly.Blazor` would be much friendlier but hard to know what to include, a more generic solution for a click handler would be great for this.

Reply:
Thanks! Well, I will just stick some screen shoots of data that came back for Pie. It seems like JavaScript did manage to return and cast to Pie from Jason. 

Labels came as ```List<object>```
![image](https://user-images.githubusercontent.com/48607649/159981135-deecfe67-df82-4d1b-9602-a8f6f5e52b9b.png)

However, values came as **NumberKind** of **JasonElement**
![image](https://user-images.githubusercontent.com/48607649/159981413-8a12074a-fec8-4ed9-befb-15cedd00637e.png)

So it seems like it did manage to deserialize Jason into some type but not entirely 🤔  But I do not think it is a big deal. We still could go and send it back as Jason then convert to string and try use ```Newtonsoft``` to parse it to some kind of object? 

At this point we would have char type and data as json string so, I guess we could try do something similar to what I am doing on Ribbon example:
```
IEnumerable<ITrace> GetRibbonsData()
    {
        var client = new HttpClient
        {
            BaseAddress = new Uri(MyNavigationManager.BaseUri)
        };
        
        var jsonResponse = client.GetStringAsync(Url).Result;
        var jasonObject = Newtonsoft.Json.Linq.JObject.Parse(jsonResponse);
        var results = jasonObject["data"].Children().ToList();

        return results.Select(result =>
        {
            var ribbon = result.ToObject<Ribbon>();
            return new Surface()
            {
                X = ribbon.X,
                Y = ribbon.Y,
                Z = ribbon.Z,
                ShowLegend = false,
                ShowScale = false
            };
        });
    }
```

 

> The design here is very open, `Plotly.Blazor` forwards all the event data, and the consuming project (e.g. Examples) decides what event handlers to define in JavaScript and C#. The example here intentionally only passed through data it was interested in to reduce the amount of data being transferred/deserialized and improve readability, but it could pass through a lot more as needed. This makes it very flexible for different charts returning different data shapes / types (e.g. is `X` a string or number) but pushes the complexity of defining them the consuming developers. Being able to define some standard click handlers for the most common cases in `Plotly.Blazor` would be much friendlier but hard to know what to include, a more generic solution for a click handler would be great for this.

I see, I think it is a great and easy to understand example to how to get this thing to work and expose these events so GJ 😄